ql/time/schedule.cpp

```diff
@@ -426,9 +433,12 @@ namespace QuantLib {
         // date due to EOM adjustments (see the Schedule test suite
         // for an example).
         if (dates_.size() >= 2 && dates_[dates_.size()-2] >= dates_.back()) {
-            isRegular_[isRegular_.size()-2] =
-                (dates_[dates_.size()-2] == dates_.back());
-            dates_[dates_.size()-2] = dates_.back();
+            // there might be two dates only, then isRegular_ has size one
+            if (isRegular_.size() >= 2) {
+                isRegular_[isRegular_.size() - 2] =
+                    (dates_[dates_.size() - 2] == dates_.back());
+            }
+            dates_[dates_.size() - 2] = dates_.back();
             dates_.pop_back();
             isRegular_.pop_back();
```

Comment:
Should this go into the `if (isRegular_.size() >= 2)`?

Reply:
I think what is done to `dates_` in the current code is fine, no matter if there are only 2 entries in `dates_` or more. But in case there are only 2 entries in `dates_`, we only have 1 entry in `isRegular_` so we can't do lines 438/439 (and don't need to, since the only entry in`isRegular_` is removed anyhow below.

So in short, no. I think.